src/librustc_codegen_ssa/mir/block.rs

```diff
@@ -566,7 +566,7 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
 
         if let Some(sfcx) = &mut self.sir_func_cx {
             let opnd = match instance {
-                Some(inst) if inst.def_id() == sfcx.trace_inputs_defid => {
+                Some(inst) if bx.tcx().symbol_name(inst).name == "__yk_trace_inputs" => {
```

Comment:
Would it be better to introduce an attribute for this so that we don't have to constantly do string comparisons?

Reply:
Done